File: levels/level7/solution.cpp
// Lösung: Implementiere den Bubble Sort Algorithmus

#include <iostream>
#include <vector>
#include <chrono>
#include <iomanip>

void bubble_sort(std::vector<int>& arr) {
    int n = arr.size();
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                std::swap(arr[j], arr[j + 1]);
            }
        }
    }
}

int main() {
    std::vector<int> arr = {64, 34, 25, 12, 22, 11, 90};
    bubble_sort(arr);
    std::cout << "Sortierte Liste: ";
    for (int x : arr) {
        std::cout << x << " ";
    }
    std::cout << std::endl;

    // Teste den Algorithmus und messe die Zeit
    std::vector<int> large_arr(1000);
    for (int i = 0; i < 1000; ++i) {
        large_arr[i] = 1000 - i;
    }

    auto start_time = std::chrono::high_resolution_clock::now();
    bubble_sort(large_arr);
    auto end_time = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double, std::milli> time_taken = end_time - start_time;

    std::cout << "\nTest mit großer Liste:" << std::endl;
    std::cout << "Sortierte Liste: ";
    for (int i = 0; i < 10; ++i) {
        std::cout << large_arr[i] << " ";
    }
    std::cout << "... ";
    for (int i = 990; i < 1000; ++i) {
        std::cout << large_arr[i] << " ";
    }
    std::cout << std::endl;
    std::cout << "Zeit für Bubble Sort: " << std::fixed << std::setprecision(3) << time_taken.count() << " ms" << std::endl;

    return 0;
}

File: levels/level7/solution.py
# Lösung: Implementiere den Bubble Sort Algorithmus

import time

def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

def measure_time(func, arr):
    start_time = time.time()
    func(arr)
    end_time = time.time()
    return (end_time - start_time) * 1000

# Teste den Bubble Sort Algorithmus
if __name__ == "__main__":
    arr = [64, 34, 25, 12, 22, 11, 90]
    bubble_sort(arr)
    print("Sortierte Liste:", arr)

    # Teste den Algorithmus und messe die Zeit
    large_arr = [i for i in range(1000, 0, -1)]
    time_taken = measure_time(bubble_sort, large_arr)
    print("\nTest mit großer Liste:")
    print("Sortierte Liste:", large_arr[:10], "...", large_arr[-10:])  # Zeige nur einen Teil der Liste
    print(f"Zeit für Bubble Sort: {format(time_taken, '.3f')} ms")
